hcl2template/types.packer_config.go

```diff
@@ -639,6 +642,47 @@ func (cfg *PackerConfig) getCoreBuildPostProcessors(source SourceUseBlock, block
 	return res, diags
 }
 
+// GetHCPPackerRegistryBlock return the HCP registry configuration block
+// that can should be used for the current build. Right now, it should
+// use the block at the top level but support the block inside the first
+// build block with a deprecation diagnostic
+func (cfg *PackerConfig) GetHCPPackerRegistry() (*HCPPackerRegistryBlock, hcl.Diagnostics) {
+	var block *HCPPackerRegistryBlock
+	var diags hcl.Diagnostics
+
+	multipleRegistryDiag := func(*HCPPackerRegistryBlock) *hcl.Diagnostic {
+		return &hcl.Diagnostic{
+			Summary:  "Multiple HCP Packer registry block declaration",
+			Subject:  block.HCL2Ref.DefRange.Ptr(),
+			Severity: hcl.DiagError,
+			Detail:   "Multiple HCP Packer Registry block has been found, please configure it at the top level",
+		}
+	}
+	// We start by looking in the build blcoks
+	for _, build := range cfg.Builds {
+		if build.HCPPackerRegistry != nil && block != nil {
+			// error multiple build block
+			diags = append(diags, multipleRegistryDiag(build.HCPPackerRegistry))
+			return nil, diags
```

Comment:
I would not return immediately in this case, just because we might have more blocks here so we should probably report them all at once instead of doing it in multiple runs if that happens.